compiler/src/codegen.rs

```diff
@@ -702,6 +742,41 @@ pub fn gen(
     results
 }
 
+#[cfg(feature = "protobufv3-codec")]
+pub fn gen(
+    file_descriptors: &[FileDescriptorProto],
+    files_to_generate: &[String],
+) -> Vec<compiler_plugin::GenResult> {
+    let files_map: HashMap<&str, &FileDescriptorProto> =
+        file_descriptors.iter().map(|f| (f.get_name(), f)).collect();
+
+    let res = protobuf_codegen::Codegen::new()
```

Comment:
Why need this? They should be generated by protobuf codegen itself.

Reply:
Indeed.